nntrainer/include/layer.h

```diff
@@ -537,11 +537,16 @@ class Layer {
    * @param[in] type property type to be passed
    * @param[in] value value to be passed, if empty string is passed, do nothing
    * but throws error when @a type is invalid
+   * @exception std::out_of_range     when property type is not valid for the
+   * particular layer
    * @exception std::invalid_argument invalid argument
    */
   virtual void setProperty(const PropertyType type,
                            const std::string &value = "");
 
+  /// @todo move virtual void setProperty to public and remove this
+  friend class NeuralNetwork;
+
```

Comment:
do you think it is necessary to make NeuralNetwork as friend?

Reply:
`virtual void setProperty` have to move to public. But it's lengthy and trivial. so I made friend for now and going to patch this  in other PR.

Reply:
Patched at #320